### ISSUE TITLE
OSError: Could not find a suitable TLS CA certificate bundle, invalid path: C:\WINDOWS\System32\cacert.pem

### ISSUE THREAD

**Issue by murpg** [Azure/azure-cli#10079]
After installing azure-cli latest on my Windows 8.1 OS laptop and trying login with these credentials. 
`az login --service-principal -u %TF_VAR_client_id% -p %TF
_VAR_client_secret% -t %TF_VAR_tenant_id%`
This seems to work on Windows 10 ProThe command failed with an unexpected error. Here is the traceback:

Could not find a suitable TLS CA certificate bundle, invalid path: C:\WINDOWS\System32\cacert.pem
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\commands\__init__.py", line 578, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\commands\__init__.py", line 636, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\commands\__init__.py", line 629, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\commands\__init__.py", line 606, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\commands\__init__.py", line 305, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\__init__.py", line 485, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli\azure\cli\command_modules\profile\custom.py", line 133, in login
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\_profile.py", line 190, in find_subscriptions_on_login
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\azure-cli-core\azure\cli\core\_profile.py", line 791, in find_through_authorization_code_flow
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\authentication_context.py", line 215, in acquire_token_with_authorization_code
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\authentication_context.py", line 128, in _acquire_token
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\authentication_context.py", line 213, in token_func
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\token_request.py", line 323, in get_token_with_authorization_code
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\token_request.py", line 112, in _oauth_get_token
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\adal\adal\oauth2_client.py", line 268, in get_token
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\api.py", line 116, in post
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\api.py", line 60, in request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\sessions.py", line 533, in request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\sessions.py", line 646, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\adapters.py", line 416, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-qdpilz60\requests\requests\adapters.py", line 228, in cert_verify
OSError: Could not find a suitable TLS CA certificate bundle, invalid path: C:\WINDOWS\System32\cacert.pem and Windows 7.

**Comment by jiasli** [comment]
It looks like `requests` is not reading from CLI's default CA bundle at `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\certifi\cacert.pem`.

 Could you run `SET` in command prompt to check if `REQUESTS_CA_BUNDLE` is set to `C:\WINDOWS\System32\cacert.pem`? If so, please remove this environment variable and retry `az login`.

**Comment by murpg** [comment]
My variable did not exist. I created it, pointing the correct pem file   
Please ensure you have network connection. Error detail: HTTPSConnectionPool(host='login.microsoftonline.com', port=443):
Max retries exceeded with url: /7f96a3b4-8d9b-42d7-8f3b-b3163f89435e/oauth2/token (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))

**Comment by jiasli** [comment]
`certificate verify failed` is usually caused by self-signed certificate. Please check if there is any proxy being used. If so, please add the PEM file of this certificate to the CA bundle. More info: https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy

**Comment by murpg** [comment]
I am not behind a proxy. I have unchecked bypass proxy server for local addresses in the IE connections tab > lan settings. I get the same error as before.

**Comment by murpg** [comment]
How can I get a cacert.pem file which I can use on that Azure will accept?

**Comment by jiasli** [comment]
**bypass proxy server for local addresses** is not related to CLI using system proxy. You may follow this link to check the certificate your proxy is using: https://www.ssl2buy.com/wiki/how-to-view-ssl-certificate-details-on-chrome-56.

**Comment by murpg** [comment]
I am NOT using a proxy. Can you please tell me how to solve this?

**Comment by jiasli** [comment]
Could you create an Azure support request and point to this issue #10079? A support engineer can help you diagnose issues of a local machine via remote session.

**Comment by jiasli** [comment]
Closing due to no update. If the issue persists, please let us know.

**Comment by ivyshaft** [comment]
I was able to get it working after a similar error on my machine. I had just updated to a new version of the CLI. Fixed it with the following, executed from PowerShell: 

$env:REQUESTS_CA_BUNDLE = "C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\certifi\cacert.pem"

I guess at one point I had the 32 bit version installed and the environment variable didn't get updated on the new install. The package I recently updated to was 64 bit. This is why I had to change my path. When it was WRONG it was the following value: 

$env:REQUESTS_CA_BUNDLE
C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\certifi\cacert.pem
